packages/babel-helpers/src/helpers.js

```diff
@@ -442,6 +442,23 @@ helpers.specRequireInterop = template(`
   })
 `);
 
+helpers.specImportCheck = template(`
+  (function (module, imports) {
+    if (!module.__esModule) throw new Error("Only ES modules can be checked");
+    const realImports = Object.keys(module);
+    const invalid = imports.filter(function (i) { return realImports.indexOf(i) < 0; });
```

Comment:
`const invalid = imports.filter(function (i) { return Object.prototype.hasOwnProperty.call(module, i); })` would be faster.